docs/probing.md

```diff
@@ -0,0 +1,318 @@
+
+```rust
+/// Updates scorer based on event and returns whether an update occurred so we can decide whether
+/// to persist.
+fn update_scorer<'a, S: 'static + Deref<Target = SC> + Send + Sync, SC: 'a + WriteableScore<'a>>(
+	scorer: &'a S, event: &Event, duration_since_epoch: Duration,
+) -> bool {
+	match event {
+		Event::PaymentPathFailed { ref path, short_channel_id: Some(scid), .. } => {
+			let mut score = scorer.write_lock();
+			score.payment_path_failed(path, *scid, duration_since_epoch);
+		},
+		Event::PaymentPathFailed { ref path, payment_failed_permanently: true, .. } => {
+			// Reached if the destination explicitly failed it back. We treat this as a successful probe
+			// because the payment made it all the way to the destination with sufficient liquidity.
+			let mut score = scorer.write_lock();
+			score.probe_successful(path, duration_since_epoch);
+		},
+		Event::PaymentPathSuccessful { path, .. } => {
+			let mut score = scorer.write_lock();
+			score.payment_path_successful(path, duration_since_epoch);
+		},
+		Event::ProbeSuccessful { path, .. } => {
+			let mut score = scorer.write_lock();
+			score.probe_successful(path, duration_since_epoch);
+		},
+		Event::ailed { path, short_channel_id: Some(scid), .. } => {
+			let mut score = scorer.write_lock();
+			score.probe_failed(path, *scid, duration_since_epoch);
+		},
+		_ => return false,
+	}
+	true
+}
+```
+
+It's important that our scorer also observes the passage of time. This is because if our scorer has not received any recent updates for a channel, it will need to broaden its current estimates for available channel liquidity. You can read more about the ```time_passed``` parameter [here](https://docs.rs/lightning/latest/lightning/routing/scoring/trait.ScoreUpdate.html#tymethod.time_passed).
+
+Similar to handling payment and probe-related events, the background processor that we created in the previous section will call ```time_passed``` on our scorer and persist the updated scorer every 5 minutes.
+
+#### Sending Probes
+
+We're almost done! Now that we have probing functionality built out and the ability to update our scorer with the results, we just need to tell our application to send a probe randomly, once per minute. 
+
+Within your application, you will likely have a main entry point where you initialize important LDK components such as the ```ChannelManager```, ```ChainMonitor```, and the ```ProbabilisticScorer``` itself. Since all of the required components will be available within this asynchronous entry point, we can configure our app to send a probe every 60 seconds by calling the functions we previously created and passing in the required components.
+
+```rust
+// Regularly probe.
+let probing_cm = Arc::clone(&channel_manager);
+let probing_graph = Arc::clone(&network_graph);
+let probing_logger = Arc::clone(&logger);
+let probing_scorer = Arc::clone(&scorer);
```

Comment:
We should use a fresh scorer here rather than the same one used for updates, IIUC.